### ISSUE TITLE
After changing to cargo chef the program stops right after execution

### ISSUE THREAD

**Issue by EvilWatermelon** [LukeMathWalker/cargo-chef#233]
I recently changed my Dockerfile to `cargo chef` and after it runs successfully then my `actix-web` program stops immediatly with `exit code 0`.  What do I have to change in my Dockerfile?

```
# Build stage
FROM lukemathwalker/cargo-chef:latest-rust-1 AS chef

RUN apt-get update && \
  apt-get install -y pkg-config make g++ libssl-dev cmake libmariadb-dev-compat && \
  rustup target add x86_64-unknown-linux-gnu 

WORKDIR /var/www/app

FROM chef AS planner
COPY . .
RUN cargo chef prepare --recipe-path recipe.json

FROM chef AS builder
COPY --from=planner /var/www/app/recipe.json recipe.json
RUN cargo chef cook --release --recipe-path recipe.json

COPY . .

RUN cargo build --release

# Prod stage, removing the Rust toolchain
FROM debian:buster-slim AS runtime
WORKDIR /var/www/app
COPY --from=builder /var/www/app/target/release/isumis /usr/local/bin

CMD ["/usr/local/bin/isumis"]
```

**Comment by lanklaas** [comment]
I also started getting this recently. I was able to create a minimal example for you [here](https://github.com/lanklaas/cargo-chef-demo).

To test I ran the build:
```bash
cd cargo-chef-demo
docker build -t test .
```

Then shell into the container to run the bin:
```bash
docker run -it test ash
./chef-builds
```

It should print "Hello,world!", but just exits.

Let me know if I can do anything to help

**Comment by EvilWatermelon** [comment]
@lanklaas it doesn't print "Hello,world!". 

```
cargo-chef-demo$ sudo docker run -it test ash
/app $ ./chef-builds
/app $ 
```

@LukeMathWalker I added it.

**Comment by lanklaas** [comment]
Sorry, think I worded it wrong. The example I gave is an example of the broken one. So it not printing `Hello,world!` is the problem. I pushed another dockerfile to the repo to show you the working example (without using cargo-chef). If you pull again you can build it like this:

```bash
docker build -f working.Dockerfile -t wtest .
```

Then run it:

```bash
docker run -it wtest ash
./chef-builds
```

This one prints `Hello,world!`

An interesting thing I noticed was that the binary file sizes differs just a little:

Broken Size:

```
/app $ ls -l
total 4372
-rwxr-xr-x    1 root     root       4475024 Aug 18 06:11 chef-builds
```

Working size

```
/app $ ls -l
total 4384
-rwxr-xr-x    1 root     root       4481912 Aug 18 06:56 chef-builds
```

**Comment by EvilWatermelon** [comment]
And you think this should solve my problem?

**Comment by dman-os** [comment]
I run into this as well. After a whole day of tinkering with things, I was able to narrow down the issue to cargo-chef. Removing it from the Dockerfile revealed the real issue, I had mistakenly excluded some necessary files through `.dockerignore` entries (including the `main.rs`). And yet, with `cargo chef cook` in the recipe, the Dockerfile compiled and completed without an issue! Fixing the ignore entries solved everything and hopefully, this case helps to narrow down the issue.

**Comment by LukeMathWalker** [comment]
@lanklaas: pretty sure the issue is due to a swapped statement in your Dockerfile, see https://github.com/lanklaas/cargo-chef-demo/pull/1
I'm not at my laptop to confirm it works, but I'm fairly confident.

**Comment by lanklaas** [comment]
@LukeMathWalker thanks, Yes I forgot I swapped those because I got version issues before the swap. I added 2 branches on the repo [correct-copy-order](https://github.com/lanklaas/cargo-chef-demo/tree/correct-copy-order) which is the same fix as your PR, and [workspace](https://github.com/lanklaas/cargo-chef-demo/tree/workspace) in which I changed the repo to a cargo workspace to fix the error in `correct-copy-order`, just in case it helps someone else here.

I tracked down the original issue that caused me to do the swap, but it is not related to this issue. I will investigate and report it if I have some more concrete info
